### PR TITLE
Upgrade to Spring Boot 3

### PR DESCRIPTION
**Pull request by pascalgrimaud**
- fix https://github.com/jhipster/jhipster-lite/issues/3601

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/module/infrastructure/primary/OpenApiModulesConfiguration.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ class OpenApiModulesConfiguration {
   private static final String JSON_MEDIA_TYPE = "application/json";
 
   @Bean
-  OpenApiCustomiser openApiModules(JHipsterModulesResources modules) {
+  OpenApiCustomizer openApiModules(JHipsterModulesResources modules) {
```
[the remark this answers is not in the document]
Comment (pascalgrimaud) [reply src/main/java/tech/jhipster/lite/module/infrastructure/primary/OpenApiModulesConfiguration.java:47]: ahahaha :D

**Comment by pascalgrimaud** [comment]
2 issues to fix:
- mongodb
- exception translator integration test

**Comment by atomfrede** [comment]
I would say let's go and merge it so see things not working early. It will also help to get the graalvm native part easier.

**Comment by DamnClin** [comment]
IMO: Go, I think JHLite is a good playground to learn on SpringBoot 3

**Comment by DamnClin** [comment]
Wait, what do you mean with this?

> fix the cucumber part too

**Comment by swarajsaaj** [comment]
Yes, I think we should go ahead with it.

**Comment by qmonmert** [comment]
Yes go 🤝

File: src/test/java/tech/jhipster/lite/module/infrastructure/primary/ModulesSteps.java [hunk 2]
```diff
@@ -325,9 +321,6 @@ public void shouldHaveFilesCountInDirectory(int filesCount, String directory) th
 
   @Then("I should have unknown slug {string} error message")
   public void shouldHaveUnknownSlugErrorMessage(String slugName) {
-    assertThatLastResponse()
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/module/infrastructure/primary/ModulesSteps.java:328]: Cc @DamnClin : I talk about this part

Comment (DamnClin) [reply src/test/java/tech/jhipster/lite/module/infrastructure/primary/ModulesSteps.java:328]: Ah, ok, no problem with the cucumber integration so, we just need arrive handling

**Comment by pascalgrimaud** [comment]
Thanks for your advices :-)

**Pull request merged**